libs/common/src/utils/hasInputChanged.ts

```diff
@@ -0,0 +1,14 @@
+import { isEqual } from 'lodash';
+import { Change } from 'firebase-functions';
+import { DocumentSnapshot } from 'firebase-functions/lib/providers/firestore';
+
+export const hasInputChanged = (
+  change: Change<DocumentSnapshot>,
+  castFn: (data: any) => any
+): boolean => {
+  if (!change.before.exists) return true; // The document was just inserted
+  if (!change.after.exists) return false; // The document is being deleted
+
+  // The input has changed since the last update
+  return !isEqual(castFn(change.after.data()), castFn(change.before.data()));
+};
```

Comment:
This is where the magic happens. We gist of it is that we *always* want to execute when the document is first inserted, never want to execute when the document is deleted, and only want to execute if the fields we care about changed during *this* update. The lodash `isEqual` does a deep equality check.